test/spec/runner/Operations.zig

```diff
@@ -0,0 +1,235 @@
+                    // try state_transition.processBlockHeader(self.pre.allocator, self.pre.cached_state, &self.op);
+                },
+                .bls_to_execution_change => {
+                    try state_transition.processBlsToExecutionChange(self.pre.cached_state, &self.op);
+                },
+                .consolidation_request => {
+                    try state_transition.processConsolidationRequest(self.pre.allocator, self.pre.cached_state, &self.op);
+                },
+                .deposit => {
+                    try state_transition.processDeposit(self.pre.allocator, self.pre.cached_state, &self.op);
+                },
+                .deposit_request => {
+                    try state_transition.processDepositRequest(self.pre.allocator, self.pre.cached_state, &self.op);
+                },
+                .execution_payload => {
+                    return error.SkipZigTest;
+                    // try state_transition.processExecutionPayload(self.pre.allocator, self.pre.cached_state, &self.op);
+                },
+                .proposer_slashing => {
+                    try state_transition.processProposerSlashing(self.pre.cached_state, &self.op, verify);
+                },
+                .sync_aggregate => {
+                    return error.SkipZigTest;
+                },
+                .voluntary_exit => {
+                    try state_transition.processVoluntaryExit(self.pre.cached_state, &self.op, verify);
+                },
+                .withdrawal_request => {
+                    try state_transition.processWithdrawalRequest(self.pre.allocator, self.pre.cached_state, &self.op);
+                },
+                .withdrawals => {
+                    var withdrawals_result = WithdrawalsResult{
+                        .withdrawals = try Withdrawals.initCapacity(
+                            self.pre.allocator,
+                            preset.MAX_WITHDRAWALS_PER_PAYLOAD,
+                        ),
+                    };
+
+                    var withdrawal_balances = std.AutoHashMap(u64, usize).init(self.pre.allocator);
+                    defer withdrawal_balances.deinit();
+
+                    try state_transition.getExpectedWithdrawals(self.pre.allocator, &withdrawals_result, &withdrawal_balances, self.pre.cached_state);
+                    defer withdrawals_result.withdrawals.deinit(self.pre.allocator);
+
+                    try state_transition.processWithdrawals(self.pre.cached_state, withdrawals_result);
+                },
+            }
+        }
+
+        pub fn runTest(self: *Self) !void {
+            if (valid) {
+                try self.process();
+                try expectEqualBeaconStates(self.post, self.pre.cached_state.state.*);
+            } else {
+                self.process() catch |err| {
+                    if (err == error.SkipZigTest) {
+                        return err;
+                    }
+                    return;
```

Comment:
Hmmm, when a test is invalid, we expect `process()` to throw an error. But it could be an issue that it is throwing an error that is different from what we are expecting, eg. we expect `attester_slashing / invalid_att1_empty_indices` to throw error because of empty indices and not because of invalid signature.

We need to provide a mechanism to debug (maybe a simple log is fine?). Here we're just naively returning and swallow that error.

Reply:
Another idea, we can separate the test setup from the actual test process by throwing some sort of `error.TestSetup` error for setup steps (eg any allocation that happens beforehand)

Reply:
Lets save this for another PR